### PR TITLE
Add new `step_indicate_na()`

### PR DESCRIPTION
**Pull request by konradsemsch**
That's a clear start from: https://github.com/tidymodels/recipes/pull/617

### CODE REVIEW

File: R/naindicate.R [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+#' @keywords datagen
+#' @concept preprocessing
+#' @concept imputation
+#' @export
+#' @examples
+#' library(modeldata)
+#' data("credit_data")
+#'
+#' ## missing data per column
+#' vapply(credit_data, function(x) mean(is.na(x)), c(num = 0))
```
Comment (topepo) [inline R/naindicate.R:33]: please use `purrr::map_dbl()` instead.

**Review by topepo — CHANGES_REQUESTED** [review]
Looks good!

**Comment by topepo** [comment]
One more thing that I forgot... please add a note to the NEWS file.

**Comment by konradsemsch** [comment]
Done, should be good to go!

**Comment by juliasilge** [comment]
@konradsemsch It seems that in the last commit f9cd5751d98f17c34c9a1e20ed337987966e95de a whole lot of unrelated files got checked in. We only want the files related to this PR checked in, not the `.Rd` files for other steps, and for example no `step_shadow_missing` methods in the NAMESPACE. Do you think you can give a go at fixing up this PR?

**Comment by konradsemsch** [comment]
I just spotted that in the R file:

```r
#' @export
tidy.step_shadow_missing <- function(x, ...) {
  if (is_trained(x)) {
```

and let me remove it, before I make the change. I think I also committed everything after loading the package locally so let me revert that.

**Comment by konradsemsch** [comment]
@juliasilge let me know if things are as you would expect and sorry about the previous push clutter :wink:

**Comment by topepo** [comment]
Are all of the changes in for this PR?

**Comment by konradsemsch** [comment]
All from my side :)

File: R/naindicate.R [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  )
+}
+
+#' @export
+bake.step_indicate_na <- function(object, new_data, ...) {
+  col_names <- object$columns
+
+  df_ind_na <- purrr::map_dfc(
+      new_data[col_names],
+      ~ifelse(is.na(.x), 1, 0)
```
Comment (topepo) [inline R/naindicate.R:120]: Can you make these integers using `ifelse(is.na(.x), 1L, 0L)`?

**Review by topepo — CHANGES_REQUESTED** [review]
Looks great. Sorry for the lag time. One minor change.

File: R/naindicate.R (hunk 2 above)
Comment (konradsemsch) [reply R/naindicate.R]: Done

**Comment by topepo** [comment]
Thanks!

**Review by topepo — APPROVED** [review]

**Pull request merged**